File: 1004-max-consecutive-ones-iii/1004-max-consecutive-ones-iii.cpp
class Solution {
public:
    int longestOnes(vector<int>& nums, int k) {
       int left = 0, right = 0, maxLength = 0, zeroCount = 0;
        int n = nums.size();
        
        while (right < n) {
            if (nums[right] == 0) {
                zeroCount++;
            }
            
            while (zeroCount > k) {
                if (nums[left] == 0) {
                    zeroCount--;
                }
                left++;
            }
            
            maxLength = std::max(maxLength, right - left + 1);
            right++;
        }
        
        return maxLength;
    }
};

File: 1004-max-consecutive-ones-iii/1004-max-consecutive-ones-iii.py
class Solution(object):
    def longestOnes(self, nums, k):
        l, r, c, ans, n = 0, 0, 0, 0, len(nums)
        while r < n:
            if nums[r] == 0: c += 1
            while c > k:
                if nums[l] == 0: c -= 1
                l += 1
            ans = max(ans, r - l + 1)
            r += 1
        return ans